src/gui/plugins/resource_spawner/ResourceSpawner.cc

```diff
@@ -0,0 +1,289 @@
+    return;
+
+  // If we have found model.config, extract thumbnail and sdf
+  LocalModel model;
+  std::string modelPath = common::parentPath(_path);
+  std::string thumbnailPath = common::joinPaths(modelPath, "thumbnails");
+  std::string configFileName = common::joinPaths(modelPath, "model.config");
+  tinyxml2::XMLDocument doc;
+  doc.LoadFile(configFileName.c_str());
+  auto modelXml = doc.FirstChildElement("model");
+
+  if (modelXml)
+  {
+    auto modelName = modelXml->FirstChildElement("name");
+    if (modelName)
+      model.name = modelName->GetText();
+  }
+  std::string sdfPath = sdf::getModelFilePath(modelPath);
+  model.sdfPath = sdfPath;
+
+  // Get first thumbnail image found
+  if (common::exists(thumbnailPath))
+  {
+    for (common::DirIter file(thumbnailPath);
+         file != common::DirIter(); ++file)
+    {
+      std::string current(*file);
+      if (common::isFile(current))
+      {
+        std::string thumbnailFileName = common::basename(current);
+        std::string::size_type thumbnailExtensionIndex =
+          thumbnailFileName.rfind(".");
+        std::string thumbnailFileExtension =
+          thumbnailFileName.substr(thumbnailExtensionIndex + 1);
+        // The standard image types QML supports
+        if (thumbnailFileExtension == "png"  ||
+            thumbnailFileExtension == "jpg"  ||
+            thumbnailFileExtension == "jpeg" ||
+            thumbnailFileExtension == "svg")
+        {
+          model.thumbnailPath = current;
+          break;
+        }
+      }
+    }
+  }
+  this->dataPtr->gridModel.AddLocalModel(model);
+}
+
+/////////////////////////////////////////////////
+void ResourceSpawner::FindLocalModels(const std::string &_path)
+{
+  std::string path = _path;
+  if (common::isDirectory(path))
+  {
+    for (common::DirIter file(path); file != common::DirIter(); ++file)
+    {
+      std::string currentPath(*file);
+      if (common::isDirectory(currentPath))
+      {
+        std::string modelConfigPath =
+          common::joinPaths(currentPath, "model.config");
+          this->LoadLocalModel(modelConfigPath);
+      }
+      else
+      {
+        this->LoadLocalModel(currentPath);
+      }
+    }
+  }
+  else
+  {
+    this->LoadLocalModel(path);
+  }
+}
+
+/////////////////////////////////////////////////
+void ResourceSpawner::AddPath(const std::string &_path)
+{
+  this->dataPtr->pathModel.AddPath(_path);
+}
+
+/////////////////////////////////////////////////
+void ResourceSpawner::OnPathClicked(const QString &_path)
+{
+  this->dataPtr->gridModel.Clear();
+  this->FindLocalModels(_path.toStdString());
+}
+
+/////////////////////////////////////////////////
+void ResourceSpawner::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Resource Spawner";
+
+  // For resource spawn requests
+  ignition::gui::App()->findChild
+    <ignition::gui::MainWindow *>()->installEventFilter(this);
```

Comment:
I don't see an `eventFilter` callback, I think this can be removed.

Reply:
Resolved 96e39b105f768e4e940c74a2e07c3118c411a73e